scripts/symbolicate-profile.ts

```diff
@@ -0,0 +1,208 @@
+            owner: OWNER,
+            repo: REPO,
+            artifact_id: artifactId,
+            archive_format: 'zip',
+        })
+        .then((response) => {
+            // The response should be a redirect to the actual download URL
+            const downloadUrl = response.url;
+
+            if (downloadUrl === undefined) {
+                throw new Error(`Could not find the download URL in:\n${JSON.stringify(response, null, 2)}`);
+            }
+
+            return downloadUrl;
+        })
+        .catch((error) => {
+            Logger.error('Failed to download artifact!');
+            Logger.error(error);
+            throw error;
+        });
+}
+
+const dirName = '.sourcemaps';
+const sourcemapDir = path.join(process.cwd(), dirName);
+
+function downloadFile(url: string) {
+    Logger.log(`Downloading file from URL: ${url}`);
+    if (!fs.existsSync(sourcemapDir)) {
+        Logger.info(`Creating download directory ${sourcemapDir}`);
+        fs.mkdirSync(sourcemapDir);
+    }
+
+    const destination = path.join(sourcemapDir, `${argsMap.platform}-sourcemap-${appVersion}.zip`);
+    const file = fs.createWriteStream(destination);
+    return new Promise<string>((resolve, reject) => {
+        https
+            .get(url, (response) => {
+                response.pipe(file);
+                file.on('finish', () => {
+                    file.close();
+                    Logger.success(`Downloaded file to ${destination}`);
+                    resolve(destination);
+                });
+            })
+            .on('error', (error) => {
+                fs.unlink(destination, () => {
+                    reject(error);
+                });
+            });
+    });
+}
+
+function unpackZipFile(zipPath: string) {
+    Logger.info(`Unpacking file ${zipPath}`);
+    const command = `unzip -o ${zipPath} -d ${sourcemapDir}`;
+    execSync(command, {stdio: 'inherit'});
+    Logger.info(`Deleting zip file ${zipPath}`);
+    return new Promise<void>((resolve, reject) => {
+        fs.unlink(zipPath, (error) => (error ? reject(error) : resolve()));
+    });
+}
+
+const localSourceMapPath = path.join(sourcemapDir, `${appVersion}-${argsMap.platform}.map`);
+function renameDownloadedSourcemapFile() {
+    const androidName = 'index.android.bundle.map';
+    const iosName = 'main.jsbundle.map';
+    const downloadSourcemapPath = path.join(sourcemapDir, argsMap.platform === 'ios' ? iosName : androidName);
+
+    if (!fs.existsSync(downloadSourcemapPath)) {
+        Logger.error(`Could not find the sourcemap file ${downloadSourcemapPath}`);
+        process.exit(1);
+    }
+
+    Logger.info(`Renaming sourcemap file to ${localSourceMapPath}`);
+    fs.renameSync(downloadSourcemapPath, localSourceMapPath);
+}
+
+// Symbolicate using the downloaded source map
+function symbolicateProfile() {
+    const command = `npx react-native-release-profiler --local ${argsMap.profile} --sourcemap-path ${localSourceMapPath}`;
```

Comment:
Suggestion, add some logs and error handling:

```ts
function symbolicateProfile() {
    Logger.info(`Symbolicating profile ${argsMap.profile} with sourcemap ${localSourceMapPath}`);
    const command = `npx react-native-release-profiler --local ${argsMap.profile} --sourcemap-path ${localSourceMapPath}`;
    Logger.info(`Executing command: ${command}`);
    try {
        execSync(command, {stdio: 'inherit'});
        Logger.success('Sourcemap symbolicated!');
    } catch (err) {
        Logger.error(`Error executing command: ${command}`, err);
    }
}
```

Reply:
I know that we did this in some other cases, but actually the command with execSync will:
- print to the console (stdio: inherit)
- if the command fails (the error will be printed by the tool itself), and our process will exit with the same exit code as the command invoked with execSync

So all this try catch would catch if something in the node internals fails to execute `execSync`, which is very unlikely.

So I felt the code would be leaner without that additional try catch

Reply:
oh nice, I missed the `{stdio: inherit}`, and it's actually a great trick I could've used in the past 😄